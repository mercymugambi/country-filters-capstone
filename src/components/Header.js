import React from 'react';
import { useSelector } from 'react-redux';
import africa from './assets/africa.png';

const Header = () => {
  const { countries } = useSelector((state) => state.countries);

  return (
    <>
      <div className="navbar">
        <span className="theme">Population Metrics</span>
      </div>
      <div className="mapSection">
        <img className="svg-container" src={africa} alt="africa" />
        <div className="totalCountries">
          <span className="tCountries">Total Countries</span>
          <br />
          <span className="main-text">{countries.length}</span>
        </div>
      </div>
    </>
  );
};

export default Header;
