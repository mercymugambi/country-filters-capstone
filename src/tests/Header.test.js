import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Header from '../components/Header';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore([]);

test('render Header component correctly', () => {
  const mockCountries = [
    { name: 'Country 1' },
    { name: 'Country 2' },
    { name: 'Country 3' },
  ];

  const store = mockStore({ countries: { countries: mockCountries } });

  const { container } = render(
    <Provider store={store}>
      <Header />
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
