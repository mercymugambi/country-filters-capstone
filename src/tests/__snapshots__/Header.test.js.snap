// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component correctly 1`] = `
<div
  class="navbar"
>
  <span
    class="theme"
  >
    Population Metrics
  </span>
</div>
`;

exports[`renders Header component correctly 1`] = `
<div
  class="navbar"
>
  <span
    class="theme"
  >
    Population Metrics
  </span>
</div>
`;
